Fix audio crackling

Gymnasiast opened the pull request:
For me, this solved the problem when resampling a 48 kHz OGG to 22.05 kHz, or a 22.05 kHz file to 48 kHz. It seems that the length was getting truncated by rounding errors.

I removed _everything_ from this PR except the fix, as it seems that people were getting distracted by the sample rate change.

Here is a test file. You can test as follows:
1. Add the object to a park
2. Add it to a Merry-go-round and play.
3. Observe there is crackling on `develop`, but not on this branch.

[gymnasiast.music.crackletest.parkobj.zip](https://github.com/user-attachments/files/23564383/gymnasiast.music.crackletest.parkobj.zip)

ZehMatt commented:
I changed this in the past and then we reverted it, since this only happens to a few people I think that the root cause has to yet discovered.

janisozaur commented on src/openrct2-ui/audio/AudioMixer.cpp:
```diff
@@ -27,10 +27,10 @@ void AudioMixer::Init(const char* device)
 
     SDL_AudioSpec want = {};
-    want.freq = 22050;
+    want.freq = 48000;
```
Shouldn't we interrogate the device for the wanted frequency? I think hardcoding it either way will result in other players starting to report audio crackling now.

Gymnasiast replied:
It works the way you describe. `want` is what we requested, but SDL will return what’s actually possible in the `have` variable, which is what we save and work with.

janisozaur commented on src/openrct2-ui/audio/AudioMixer.cpp:
```diff
@@ -27,10 +27,10 @@ void AudioMixer::Init(const char* device)
     want.format = AUDIO_S16SYS;
     want.channels = 2;
-    want.samples = 2048;
+    want.samples = 8192;
```
ditto

Gymnasiast commented:
> > > I changed this in the past and then we reverted it, since this only happens to a few people I think that the root cause has to yet discovered.
> > 
> > 
> > I guess I could have explained myself better here, but: I could reproduce the bug -- it happens when it needs to resample audio -- fixed it, and only after _that_ raised the sampling rate, verifying it didn’t cause problems.
> 
> That is exactly what I did back then

I took a look through Git, even going back trough 2017, and it doesn’t seem like you ever applied [this fix](https://github.com/OpenRCT2/OpenRCT2/pull/25519/commits/248e07df2d76580fd3b6b6dd3ad3935593492740). That line has been essentially unchanged since 2017, with the only changes being code style, replacing C-style casts, etcetera. Am I missing something?

ZehMatt commented:
> I'm aware of that PR and the history of the sample rate. I was also in an audio call about it. Please don't assume I'm just reapplying the same nonworking change from 2 years ago.
> 
> As I've tried to explain: I was actually able to reproduce the issue, fixed it, and increased the rate afterwards since it was now safe to do. Since you got hung up on the sample rate changes I removed those from the PR, but even now we're still arguing. Please take a look at the now one-line diff, as well as the test file.

I'm not arguing about anything, pointing out what was done previously and that there might be more than the line you changed, nothing more.

janisozaur commented:
I was able to reproduce the issue, checked how SDL handles the code and can confirm the fix, good job @Gymnasiast ! This is ceratinly changelog-worthy
